### PR TITLE
Drag and drop improvements

### PR DESCRIPTION
**Pull request by belgamo**
- Optimistically update
When the user dragged and dropped a story, it returned to the previous position while backend was in proccess. Then, only when the response from backend was provided, the story was placed on the correct position. Now, the story will only return to the previous column if an error occurs.

Before:
ezgif com-video-to-gif

After:
ezgif com-video-to-gif(1)

- Allow drop story anywhere on the bottom of the column

Before:
ezgif com-video-to-gif(2)

After:
ezgif com-video-to-gif(3)

### CODE REVIEW

File: app/assets/javascripts/components/story/CollapsedStory/index.js [hunk 1]
```diff
@@ -41,49 +42,61 @@ export const Container = ({
...
-  >
-    <StoryPopover story={story}>
-      <div className='Story__icons-block'>
-        <StoryIcon storyType={story.storyType} />
-        <CollapsedStoryEstimate estimate={story.estimate} />
-        <StoryDescriptionIcon description={story.description} />
-      </div>
-    </StoryPopover>
+}) => {
+  const isLoading = Boolean(story.loading);
```
Comment (talyssonoc) [inline app/assets/javascripts/components/story/CollapsedStory/index.js]: Why is this explicit coercion necessary?

File: spec/javascripts/actions/story_spec.js [hunk 2]
```diff
@@ -283,11 +285,12 @@ describe('Story Actions', () => {
 
       await Story.dragDropStory(story.id, story.projectId, {
         position: 3.54,
-      })(fakeDispatch, fakeGetState, { Story: FakeStory });
+      }, from)(fakeDispatch, fakeGetState, { Story: FakeStory });
+
+      await wait(300);
```
Comment (talyssonoc) [inline spec/javascripts/actions/story_spec.js]: We shouldn't have to wait it in  the test :thinking: , why was this necessary?

**Comment by talyssonoc** [comment]
Let's also make it impossible to expand a story while it's loading :smile:

File: app/assets/javascripts/components/story/CollapsedStory/index.js (hunk 1 above)
Comment (belgamo) [reply app/assets/javascripts/components/story/CollapsedStory/index.js]: Truth be told, it wasn't necessary at all. Thanks!

File: spec/javascripts/actions/story_spec.js (hunk 2 above)
Comment (belgamo) [reply spec/javascripts/actions/story_spec.js]: Probably because I wanted simulate exactly the same behavior of real action. But this isn't really necessary. Thank you again!

File: spec/javascripts/actions/story_spec.js [hunk 3]
```diff
@@ -1,5 +1,6 @@
 import * as Story from 'actions/story';
 import storyFactory from '../support/factories/storyFactory';
+import { wait } from '../../../app/assets/javascripts/services/promises'
```
Comment (talyssonoc) [inline spec/javascripts/actions/story_spec.js]:
```suggestion
```
